cocos/gfx/webgl2/webgl2-swapchain.ts

```diff
@@ -180,20 +180,22 @@ export class WebGL2Swapchain extends Swapchain {
     private _blitManager: IWebGL2BlitManager | null = null;
 
     public initialize (info: Readonly<SwapchainInfo>): void {
-        this._canvas = info.windowHandle;
+        const self = this;
+        self._canvas = info.windowHandle;
 
-        this._webGL2ContextLostHandler = this._onWebGLContextLost.bind(this);
-        this._canvas.addEventListener(eventWebGLContextLost, this._onWebGLContextLost);
+        self._webGL2ContextLostHandler = self._onWebGLContextLost.bind(self);
+        self._canvas.addEventListener(eventWebGLContextLost, self._onWebGLContextLost);
 
-        const gl = WebGL2DeviceManager.instance.gl;
+        const { instance } = WebGL2DeviceManager;
+        const { gl, capabilities } = instance;
 
-        this.stateCache.initialize(
-            WebGL2DeviceManager.instance.capabilities.maxTextureUnits,
-            WebGL2DeviceManager.instance.capabilities.maxUniformBufferBindings,
-            WebGL2DeviceManager.instance.capabilities.maxVertexAttributes,
+        self.stateCache.initialize(
+            capabilities.maxTextureUnits,
```

Comment:
Avoid to invoke `WebGL2DeviceManager.instance.capabilities ` many times which could not be mangled.